code/obj/machinery/vortex_wraith.dm

```diff
@@ -0,0 +1,163 @@
+								eligible_turf += T
+				if (length(eligible_turf) <= 0)	//No spot to spawn near a human, or no human in range, lets try to find a regular turf instead
+					for (var/turf/T in range((1 + src.growth), src))
+						if (isturf(T) && istype(T, /turf/simulated/floor))	//Find a non-wall, non-space turf to spawn in
+							eligible_turf += T
+				if (length(eligible_turf) <= 0)
+					src.visible_message("<span class='alert'><b>[src] sputters and crackles, it seems it couldnt find a spot to summon something!</b></span>")
+					return 1
+				chosen_turf = pick(eligible_turf)
+				var/obj/decal/harbinger_portal/portal = new /obj/decal/harbinger_portal
+				portal.set_loc(chosen_turf)
+				portal.alpha = 0
+				animate(portal, alpha=255, time=1 SECONDS)
+				portal_light = new /datum/light/point
+				portal_light.set_brightness(0.1)
+				portal_light.set_color(150, 40, 40)
+				portal_light.attach(portal)
+				portal_light.enable()
+				playsound(chosen_turf, "sound/effects/flameswoosh.ogg" , 80, 1)
+				SPAWN(3 SECOND)
+					animate(portal, alpha=0, time=1 SECONDS)
+					SPAWN(1 SECOND)
+						qdel(portal_light)
+						qdel(portal)
+					if (src.random_mode)
+						src.mob_type = pick(src.default_mobs)
+					minion_value = getMobValue(src.mob_type)
+					if ((src.total_mob_value + minion_value) <= src.mob_value_cap)
+						var/obj/minion = new src.mob_type(chosen_turf)
+						src.mob_list += minion
+						minion.alpha = 0
+						animate(minion, alpha=255, time = 2 SECONDS)
+						src.visible_message("<span class='alert'><b>[minion] emerges from the [src]!</b></span>")
+						src.total_mob_value += minion_value
+			next_spawn = world.time + (20 SECONDS) + (minion_value * 5) SECONDS
+
+	attackby(obj/item/P as obj, mob/living/user as mob)
+		src._health -= P.force
+		attack_particle(user,src)
+		user.lastattacked = src
+		hit_twitch(src)
+		playsound(src.loc, 'sound/impact_sounds/Metal_Hit_Light_1.ogg', 50, 1)
+		if(src._health <= 0)
+			if (src.master != null)
+				src.master.linked_portal = null
+			deleteLinkedMobs()
+			qdel(src)
+
+	onDestroy()
+		. = ..()
+		if (src.master != null)
+			src.master.linked_portal = null
+		deleteLinkedMobs()
+
+	disposing()
+		if (src.master != null)
+			src.master.linked_portal = null
+		deleteLinkedMobs()
+		. = ..()
+
+	proc/getMobValue(var/obj/O)
+		switch (O)
+			if (/obj/critter/bear)
+				return 10
+			if (/obj/critter/wraithskeleton)
+				return 4
+			if (/obj/critter/shade)
+				return 4
+			if (/obj/critter/crunched)
+				return 4
+			if (/obj/critter/bat/buff)
+				return 3
+			if (/obj/critter/lion)
+				return 5
+			if (/obj/critter/brullbar)
+				return 15
+			if (/obj/critter/gunbot/heavy)
+				return 15
+			if (/obj/critter/ancient_thing)
+				return 7
+			if (/obj/critter/mechmonstrositycrawler)
+				return 4
+			else	//You never know, lets give an average point cost
+				return 6
+
+	proc/deleteLinkedMobs()
+		for (var/obj/C in src.mob_list)
```

Comment:
obj? 🤨

Reply:
Yes, it deletes the linked obj/critters that it spawns. They havent been transformed into the mob type yet.

Reply:
then name it critter list or something

Reply:
also fully path it to critter

Reply:
Done so!